File: blobs/howto/dotnet/BlobDevGuideBlobs/LeaseContainer.cs
using Azure;
using Azure.Storage.Blobs;
using Azure.Storage.Blobs.Models;
using Azure.Storage.Blobs.Specialized;

namespace BlobDevGuideBlobs
{
    class LeaseContainer
    {
        // <Snippet_AcquireContainerLease>
        public static async Task<BlobLeaseClient> AcquireContainerLeaseAsync(
            BlobContainerClient containerClient)
        {
            // Get a BlobLeaseClient object to work with a container lease
            BlobLeaseClient leaseClient = containerClient.GetBlobLeaseClient();

            Response<BlobLease> response =
                await leaseClient.AcquireAsync(duration: TimeSpan.FromSeconds(30));

            // Use response.Value to get information about the container lease

            return leaseClient;
        }
        // </Snippet_AcquireContainerLease>

        // <Snippet_RenewContainerLease>
        public static async Task RenewContainerLeaseAsync(
            BlobContainerClient containerClient,
            string leaseID)
        {
            // Get a BlobLeaseClient object to work with a container lease
            BlobLeaseClient leaseClient = containerClient.GetBlobLeaseClient(leaseID);

            await leaseClient.RenewAsync();
        }
        // </Snippet_RenewContainerLease>

        // <Snippet_ReleaseContainerLease>
        public static async Task ReleaseContainerLeaseAsync(
            BlobContainerClient containerClient,
            string leaseID)
        {
            // Get a BlobLeaseClient object to work with a container lease
            BlobLeaseClient leaseClient = containerClient.GetBlobLeaseClient(leaseID);

            await leaseClient.ReleaseAsync();
        }
        // </Snippet_ReleaseContainerLease>

        // <Snippet_BreakContainerLease>
        public static async Task BreakContainerLeaseAsync(
            BlobContainerClient containerClient)
        {
            // Get a BlobLeaseClient object to work with a container lease
            BlobLeaseClient leaseClient = containerClient.GetBlobLeaseClient();

            await leaseClient.BreakAsync();
        }
        // </Snippet_BreakContainerLease>
    }
}

File: blobs/howto/TypeScript/NodeJS-v12/dev-guide/src/lease-container.ts
// Import dependencies
import {
  BlobServiceClient,
  BlobLeaseClient,
  ContainerClient
} from '@azure/storage-blob';
import { DefaultAzureCredential } from '@azure/identity';
import * as dotenv from 'dotenv';

// Load environment variables from .env file
dotenv.config();

// Replace with your actual storage account name
const accountName = "<storage-account-name>";

async function main(): Promise<void> {
  // Create service client from DefaultAzureCredential
  const blobServiceClient = new BlobServiceClient(
    `https://${accountName}.blob.core.windows.net`,
    new DefaultAzureCredential()
  );

  const blobContainerClient = blobServiceClient
    .getContainerClient("sample-container");

  // Acquire lease
  const leaseClient = await acquireContainerLeaseAsync(blobContainerClient);
  // Output the blob lease status for testing
  console.log((await blobContainerClient.getProperties()).leaseState);

  // Renew lease
  await renewContainerLeaseAsync(blobContainerClient, leaseClient.leaseId);

  // Release lease
  await releaseContainerLeaseAsync(blobContainerClient, leaseClient.leaseId);
  // Output the blob lease status for testing
  console.log((await blobContainerClient.getProperties()).leaseState);

  // Break lease
  //await breakContainerLeaseAsync(blobClient);
}

// <Snippet_AcquireContainerLease>
async function acquireContainerLeaseAsync(blobContainerClient: ContainerClient) {
  const leaseClient: BlobLeaseClient = blobContainerClient.getBlobLeaseClient();
  await leaseClient.acquireLease(30);
  return leaseClient;
}
// </Snippet_AcquireContainerLease>

// <Snippet_RenewContainerLease>
async function renewContainerLeaseAsync(blobContainerClient: ContainerClient, leaseID: string) {
  const leaseClient: BlobLeaseClient = blobContainerClient.getBlobLeaseClient(leaseID);
  await leaseClient.renewLease();
}
// </Snippet_RenewContainerLease>

// <Snippet_ReleaseContainerLease>
async function releaseContainerLeaseAsync(blobContainerClient: ContainerClient, leaseID: string) {
  const leaseClient: BlobLeaseClient = blobContainerClient.getBlobLeaseClient(leaseID);
  await leaseClient.releaseLease();
}
// </Snippet_ReleaseContainerLease>

// <Snippet_BreakContainerLease>
async function breakContainerLeaseAsync(blobContainerClient: ContainerClient, breakPeriod: number) {
  const leaseClient: BlobLeaseClient = blobContainerClient.getBlobLeaseClient();
  await leaseClient.breakLease(breakPeriod);
}
// </Snippet_BreakContainerLease>

main()
  .then(() => console.log('success'))
  .catch((err: unknown) => {
    if (err instanceof Error) {
      console.log(err.message);
    }
  });